### PR TITLE
Saves some free lag by removing some in area (in world) loops

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Goes through and changes some `in area` / `in a` loops to use `get_contained_turfs` to cut down on `in_world` loops. Saves some free lag. 

## Changelog

:cl: Melbert
fix: Some things which affect everything in an area are less laggy, the "all lights are broken" station trait especially
/:cl:

### CODE REVIEW

File: code/game/machinery/lightswitch.dm [hunk 1]
```diff
@@ -87,9 +87,10 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/light_switch, 26)
 	area.lightswitch = status
 	area.update_appearance()
 
-	for(var/obj/machinery/light_switch/light_switch in area)
-		light_switch.update_appearance()
-		SEND_SIGNAL(light_switch, COMSIG_LIGHT_SWITCH_SET, status)
+	for(var/turf/area_turf as anything in area.get_contained_turfs())
+		for(var/obj/machinery/light_switch/light_switch in area_turf)
+			light_switch.update_appearance()
+			SEND_SIGNAL(light_switch, COMSIG_LIGHT_SWITCH_SET, status)
```
Comment (LemonInTheDark) [inline code/game/machinery/lightswitch.dm:93]: might actually be faster to just filter for switches in the machines list, worth profiling? how much does this actually cost

File: code/game/objects/items/blueprints.dm [hunk 2]
```diff
@@ -248,7 +248,9 @@
 		/obj/machinery/power/apc,
 	)
 
-	for(var/obj/machine as anything in area)
-		if(is_type_in_list(machine, to_rename))
+	for(var/turf/area_turf as anything in area.get_contained_turfs())
+		for(var/obj/machine in area_turf)
+			if(!is_type_in_list(machine, to_rename))
+				continue
 			machine.name = replacetext(machine.name, oldtitle, title)
```
Comment (LemonInTheDark) [inline code/game/objects/items/blueprints.dm:255]: similar

File: code/modules/power/apc/apc_main.dm [hunk 3]
```diff
@@ -463,10 +463,11 @@
...
 			emergency_lights = !emergency_lights
-			for(var/obj/machinery/light/L in area)
-				if(!initial(L.no_low_power)) //If there was an override set on creation, keep that override
-					L.no_low_power = emergency_lights
-					INVOKE_ASYNC(L, TYPE_PROC_REF(/obj/machinery/light/, update), FALSE)
+			for(var/turf/area_turf as anything in area.get_contained_turfs())
+				for(var/obj/machinery/light/area_light in area_turf)
+					if(!initial(area_light.no_low_power)) //If there was an override set on creation, keep that override
+						area_light.no_low_power = emergency_lights
+						INVOKE_ASYNC(area_light, TYPE_PROC_REF(/obj/machinery/light/, update), FALSE)
```
Comment (LemonInTheDark) [inline code/modules/power/apc/apc_main.dm:470]: applies to a lot of these actually

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/game/machinery/lightswitch.dm (hunk 1 above)
Comment (MrMelbert) [reply code/game/machinery/lightswitch.dm:93]: image

Comment (MrMelbert) [reply code/game/machinery/lightswitch.dm:93]: image
image
image

File: code/modules/power/apc/apc_main.dm (hunk 3 above)
Comment (MrMelbert) [reply code/modules/power/apc/apc_main.dm:470]: It pretty close but it seems more common machines (APCs, lights) are slower

Light switches(above)

APCs
image

Lights 
image

**Review by Ghommie — APPROVED** [review]

**Pull request merged**